src/components/Select/Select.tsx

```diff
@@ -206,73 +207,89 @@ export function Select(props: SelectProps): React.ReactElement {
         </React.Fragment>
       )
   }
+  const tooltipContent = item?.label ? (
+    <div className={css.tooltipContainer} color="white">
+      {item.label}
+    </div>
+  ) : (
+    ''
+  )
   return (
-    <Suggest
-      inputValueRenderer={opt => opt.label}
-      itemRenderer={(item: SelectOption, props: IItemRendererProps) =>
-        itemRenderer?.(item, props) || defaultItemRenderer(item, props, size)
-      }
-      itemListPredicate={(query, items) =>
-        items.filter(item => item.label.toString().toLowerCase().includes(query.toLowerCase()))
-      }
-      createNewItemFromQuery={props.createNewItemFromQuery || createNewItemFromQuery}
-      createNewItemRenderer={props.createNewItemRenderer || createNewItemRenderer}
-      noResults={<NoMatch />}
-      {...rest}
-      inputProps={{
-        onChange(e: React.ChangeEvent<HTMLInputElement>) {
-          setQuery(e.target.value)
-        },
-        value: query,
-        placeholder: Utils.getSelectComponentPlaceholder(rest?.inputProps?.placeholder),
-        leftElement: item?.icon ? <Icon size={getIconSizeFromSelect(size)} {...item?.icon} /> : undefined,
-        rightElement: (
-          <>
-            {!props.disabled && showClearBtn ? (
+    <Popover
+      boundary="viewport"
+      interactionKind="hover"
+      content={tooltipContent}
```

Comment:
If the content is undefined, do we need <Popover .../>

Reply:
Popover internally handles it on the basis of content whether to render anything on UI or not. It's a question of putting a check either on Popover component OR on content. This seemed cleaner

Reply:
Just make sure it is not adding any extra div, or wrapping children in it

Reply:
we should avoid unnecessary increases in the DOM tree.

Reply:
added a prop—now a `<Popover>` will only render if the prop `addTooltip` is true